File: api/src/main/kotlin/com/api/controller/UserController.java
package com.api.controller;

import com.knf.dev.demo.crudapplication.exception.ResourceNotFoundException;
import com.knf.dev.demo.crudapplication.model.User;
import com.knf.dev.demo.crudapplication.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.HashMap;
import java.util.List;
import java.util.Map;


@CrossOrigin(origins = "*")
@RestController
@RequestMapping("/api/v1")
public class UserController {
    @Autowired
    private UserRepository userRepository;

    @GetMapping("/users")
    public List<User> getAllUsers() {
        return userRepository.findAll();
    }

    @GetMapping("/users/{id}")
    public ResponseEntity<User> getUserById(@PathVariable(value = "id")
                                            String id) throws ResourceNotFoundException {

        User user = userRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException
                        ("User not found for this id :: " + id));
        return ResponseEntity.ok().body(user);
    }

    @PostMapping("/users")
    public User createUser(@RequestBody User user) {
        return userRepository.save(user);
    }

    @PutMapping("/users/{id}")
    public ResponseEntity<User> updateUser(@PathVariable(value = "id")
                                           String id, @RequestBody User userDto)
            throws ResourceNotFoundException {

        User user = userRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException
                        ("User not found for this id :: " + id));

        user.setEmailId(userDto.getEmailId());
        user.setLastName(userDto.getLastName());
        user.setFirstName(userDto.getFirstName());
        user.setId(id);
        final User updateUser = userRepository.save(user);
        return ResponseEntity.ok(updateUser);
    }

    @DeleteMapping("/users/{id}")
    public Map<String, Boolean> deleteUser(@PathVariable(value = "id")
                                           String id) throws ResourceNotFoundException {
        User user = userRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException
                        ("User not found for this id :: " + id));

        userRepository.delete(user);
        Map<String, Boolean> response = new HashMap<>();
        response.put("deleted", Boolean.TRUE);
        return response;
    }
}

File: api/src/main/kotlin/com/api/controller/UserController.kt
package com.api.controller

import com.core.component.exception.CommonException
import com.core.component.exception.CommonExceptionType
import com.core.entity.User
import com.core.repository.UserRepository
import lombok.RequiredArgsConstructor
import org.springframework.web.bind.annotation.*
import org.springframework.http.ResponseEntity as ResponseEntity

@CrossOrigin(origins = ["*"])
@RequiredArgsConstructor
@RestController
@RequestMapping("/api/v1")
class UserController(
    private val userRepository: UserRepository
) {


    @GetMapping("/users")
    fun allUsers(): List<User> {
        return userRepository.findAll().toList()
    }


    @GetMapping("/users/{id}")
    @Throws(CommonException::class)
    fun getUserById(@PathVariable(value = "id") id: String): ResponseEntity<User> {
        val user: User = userRepository.findById(id)
            .orElseThrow { CommonException(CommonExceptionType.NOT_FOUND_MESSAGE, "User not found for this id :: $id") }
        return ResponseEntity.ok(user)
    }

    @PostMapping("/users")
    fun createUser(@RequestBody user: User): ResponseEntity<User> {
        return ResponseEntity.ok().body(userRepository.save(user))
    }

    @PutMapping("/users/{id}")
    @Throws(CommonException::class)
    fun updateUser(@PathVariable(value = "id") id: String, @RequestBody userDto: User): ResponseEntity<User> {
        val user: User = userRepository.findById(id)
            .orElseThrow { CommonException(CommonExceptionType.NOT_FOUND_MESSAGE, "User not found for this id :: $id") }

        user.emailId = userDto.emailId
        user.lastName = userDto.lastName
        user.firstName = userDto.firstName
        user.id = id
        val updateUser = userRepository.save(user)
        return ResponseEntity.ok(updateUser)
    }

    @DeleteMapping("/users/{id}")
    @Throws(CommonException::class)
    fun deleteUser(@PathVariable(value = "id") id: String): ResponseEntity<Boolean> {
        val user: User = userRepository.findById(id)
            .orElseThrow { CommonException(CommonExceptionType.NOT_FOUND_MESSAGE, "User not found for this id :: $id") }
        userRepository.delete(user)
        return ResponseEntity.ok(true)
    }
}
